### PR TITLE
feat(plugin): allowing referer-restriction to dynamically adapt to the BE rules

### LINKED ISSUE #1588: Optimize front-end validation of fields
## The current front-end approach
To create a route for example, the front-end will be regular [1] check for certain fields, such as name field, if the regular condition is not met, the front-end page will report an error and block the user's next operation.

image

## The disadvantages of the current approach
The current front-end can meet the verification requirements. However, when the back-end changes the verification rules, the front-end has to passively and manually update the verification rules for the corresponding fields. If the back-end does not notify the front-end in time or misses some fields, then the synchronization of the front-end and back-end verification will be a problem, affecting the user's use.

## Improvement measures
After communicating with @membphis , using jsonschema validation would be a good choice.
The front-end calls the API to get the jsonschema of the corresponding module and then maps the corresponding jsonschema rules to the corresponding field checks on the front-end. Do this when the back-end update the validation rules, the front-end can automatically synchronize the latest validation rules, reducing unnecessary development time of the front-end.

[1] regular expressions provided by the back-end, and then manually added by the front-end developers to the corresponding form form for the field validation

cc @juzhiyuan @liuxiran @guoqqqi

### PR DESCRIPTION
**Pull request by wmdmomo**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

The PR will change the plugin under plugin/UI (referer-restriction), allowing referer-restriction to dynamically adapt to the back-end json rules.

**Related issues**

fix/resolve #1588 

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Review by LiteSun — APPROVED** [review]
LGTM

File: web/src/components/Plugin/UI/referer-restriction.tsx [hunk 1]
```diff
@@ -86,7 +90,7 @@ const RefererRestriction: React.FC<Props> = ({ form }) => {
                       </Form.Item>
                     </Col>
                     <Col style={{ ...removeBtnStyle, marginLeft: -10 }}>
-                      {fields.length > 1 ? (
+                      {fields.length > whiteMinLen ? (
```
Comment (juzhiyuan) [inline web/src/components/Plugin/UI/referer-restriction.tsx:93]:
```suggestion
                      {fields.length > allowListMinLength ? (
```

Comment (wmdmomo) [reply web/src/components/Plugin/UI/referer-restriction.tsx:93]: Thanks for your suggestion, I have modified it.

**Comment by liuxiran** [comment]
Test cases are needed to ensure the feature worked properly

**Comment by LiteSun** [comment]
> Test cases are needed to ensure the feature worked properly

When you have time, could you please help her with the steps of the test? @liuxiran

**Comment by liuxiran** [comment]
> > Test cases are needed to ensure the feature worked properly
> 
> When you have time, could you please help her with the steps of the test? @liuxiran

Ok, after check the current test case about referer-restriction form, @wmdmomo  can try to add test cases to https://github.com/apache/apisix-dashboard/blob/master/web/cypress/integration/route/create-route-with-referer-restriction-form.spec.js,

1. enable `referer-restriction` plugin, check default white list item should be one, and there is no  delete button behind it
2. test the white list input validate rule: input `apisix-dashboard_1.com` should be passed, and input `qq@` should be invalid, 
3. default `bypass_missing` is disabled
4. add a white list item field, each of them should have delete button behind them. 

thanks a lot

**Comment by liuxiran** [comment]
Hi @wmdmomo if you have any problem please let me know, thanks a lot

**Comment by wmdmomo** [comment]
@liuxiran Sorry to reply now. The referer-restriction test case has been completed.

**Comment by bzp2010** [comment]
@wmdmomo The code looks good to me. But please pay attention to the errors in CI.

**Comment by wmdmomo** [comment]
> @wmdmomo The code looks good to me. But please pay attention to the errors in CI.

@bzp2010 OK, thanks for your suggestion. The error in CI has been dealt with.

**Review by juzhiyuan — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]
LGTM

**Comment by bzp2010** [comment]
@wmdmomo Thank you very much for your contribution. XD

**Pull request merged**